File: frontend/src/types/pipeline.ts
export type PipelineStage = 'ingestion' | 'extraction' | 'search' | 'ranking' | 'review' | 'send';
export type PipelineStatus = 'pending' | 'in_progress' | 'completed' | 'failed';

export interface PipelineRun {
  id: number;
  transcript_id: number;
  current_stage: PipelineStage;
  status: PipelineStatus;
  ingestion_completed_at: string | null;
  extraction_completed_at: string | null;
  search_completed_at: string | null;
  ranking_completed_at: string | null;
  review_completed_at: string | null;
  send_completed_at: string | null;
  error_message: string | null;
  created_at: string;
}

File: backend/app/schemas/pipeline.py
from __future__ import annotations

from typing import TYPE_CHECKING

from pydantic import BaseModel

if TYPE_CHECKING:
    from datetime import datetime


class PipelineRunResponse(BaseModel):
    id: int
    transcript_id: int
    current_stage: str
    status: str
    ingestion_completed_at: datetime | None
    extraction_completed_at: datetime | None
    search_completed_at: datetime | None
    ranking_completed_at: datetime | None
    review_completed_at: datetime | None
    send_completed_at: datetime | None
    error_message: str | None
    created_at: datetime

    model_config = {"from_attributes": True}
